alibi_detect/saving/saving.py

```diff
@@ -302,7 +302,7 @@ def _save_preprocess_config(preprocess_fn: Callable,
     if 'preprocess_drift' in func:
         preprocess_cfg.update(kwargs)
     else:
-        kwargs.update({'kwargs': kwargs})
+        preprocess_cfg.update({'kwargs': kwargs})
```

Comment:
This is bug fix 2. Previously the `kwargs` were updated but this had no effect. It should have been the returned `preprocess_cfg` dict that was updated instead.